internal/switchboard/targets.go

```diff
@@ -24,24 +30,85 @@ type Target interface {
 //-------------------------------------------------------------------------------------------------
 
 type serviceTarget struct {
-	name types.NamespacedName
+	name   types.NamespacedName
+	logger *zap.Logger
 }
 
 // NewServiceTarget creates a new target which dynamically sources the IP from the provided
 // Kubernetes service.
-func NewServiceTarget(name, namespace string) Target {
+func NewServiceTarget(name, namespace string, log *zap.Logger) Target {
 	return serviceTarget{
-		name: types.NamespacedName{Name: name, Namespace: namespace},
+		logger: log,
+		name:   types.NamespacedName{Name: name, Namespace: namespace},
 	}
 }
 
-func (t serviceTarget) Targets(ctx context.Context, client client.Client) ([]string, error) {
+func (t serviceTarget) Targets(ctx context.Context, client client.Client, targets *string) ([]string, error) {
 	// Get service
-	var service v1.Service
-	if err := client.Get(ctx, t.name, &service); err != nil {
-		return nil, fmt.Errorf("failed to query service: %w", err)
+	if targets == nil {
+		tmp := fmt.Sprintf("%s/%s", t.name.Namespace, t.name.Name)
+		targets = &tmp
+	}
+	out := []string{}
+	for _, target := range strings.Split(*targets, ",") {
+		target = strings.TrimSpace(target)
+		if len(target) == 0 {
+			continue
+		}
+		if ipaddress.Parse(target).IsOk() {
+			t.logger.Debug(fmt.Sprintf("TargetIP:%s", target))
```

Comment:
Two things here:

- You can just use `zeus.Logger(ctx)` instead of actively passing the logger around. That's the whole purpose of `zeus`.
- `zap` logging is oriented towards structured logging. You typically have a short sentence what the log is for (e.g. "successfully parsed IP address") and then arbitrarily many key-value tuples (in this case `targetIP=<target>`, translates to `zap.String("targetIP", target)`.

Reply:
You do pass around the logger also --- so i'm not sure whats the right way.

I try to change the log.